ukat/mapping/phase_contrast.py

```diff
@@ -0,0 +1,221 @@
+"""
+
+import os
+import csv
+import numpy as np
+import nibabel as nib
+from ukat.utils.tools import convert_to_pi_range
+
+
+class PhaseContrast:
+    """
+    Generates velocity and flow measurements of an MRI Phase Contrast Sequence.
+
+    Attributes
+    ----------
+    velocity_array : np.ndarray
+        The input velocity array masked.
+    shape : tuple
+        The shape of the velocity array.
+    pixel_spacing : tuple
+        The pixel spacing of the acquisition estimated from the affine.
+    mask : np.ndarray
+        A boolean mask of the voxels to fit.
+    num_pixels_phase : list
+        List containing the number of True values in the mask per phase.
+    area_phase : list
+        List containing the area (cm2) of the mask per phase.
+    min_velocity_phase : list
+        List containing the minimum velocity values (cm/s) per phase.
+    mean_velocity_phase : list
+        List containing the average velocity values (cm/s) per phase.
+    max_velocity_phase : list
+        List containing the maximum velocity values (cm/s) per phase.
+    peak_velocity_phase : list
+        List containing the maximum velocity values (cm/s) per phase.
+    std_velocity_phase : list
+        List containing the std dev of the velocity values (cm/s) per phase.
+    RBF : list
+        List containing the Renal Blood Flow values (ml/min) per phase.
+    stats_table : dictionary
+        A dictionary containing all class attributes that are a list.
+    mean_velocity : float
+        Average velocity (cm/s) accross the different phases.
+    mean_RBF : float
+        Average Renal Blood Flow (ml/min) accross the different phases.
+    resistive_index : float
+        A prognostic marker in renal vascular diseases which range is [0, 1].
+    """
+
+    def __init__(self, velocity_array, affine, mask=None):
+        """Initialise a PhaseContrast class instance.
+
+        Parameters
+        ----------
+        velocity_array : np.ndarray
+            A 3D array containing the velocity images of the phase contrast
+            sequence, i.e. the dimensions of the array are [x, y, p], where p
+            corresponds to the phase (or trigger delay).
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        mask : np.ndarray, optional
+            A boolean mask of the voxels to fit. Should be the shape of the
+            the raw data.
+        """
+        self.shape = velocity_array.shape
+        self.affine = affine
+        self.pixel_spacing = (np.linalg.norm(self.affine[:3, 1]),
+                              np.linalg.norm(self.affine[:3, 0]))
+        # Generate a mask if there isn't one specified
+        if mask is None:
+            self.mask = np.ones(self.shape, dtype=bool)
+        else:
+            self.mask = np.where(mask == 0, np.nan, mask)
+            # The purpose is for the np.nanmean to work properly later
```

Comment:
Comment above the line rather than below it. Also, could it be rephrased a bit? I may have just been slow but I had to play around in a terminal to understand what this was doing. Suggest something along the lines of `Set masked areas to np.nan to enable differentiation between masked voxels and voxels where data is zero using np.nanmean`